### DISCUSSION TITLE
How to add audio data at the corresponding stage when the peak has already been pre rendered

### DISCUSSION THREAD

**Discussion by code565** [katspaugh/wavesurfer.js#2820] [General]
getPeaks().then(res => {
        wavesurfer.backend.setPeaks(res.data, wavesurfer.getDuration()); 
        if (wavesurfer.peakCache !== null) { 
          wavesurfer.peakCache.clearPeakCache(); 
        } 
        wavesurfer.drawBuffer();
        fetch('http://localhost:8089/api/wav?index=0')
        .then(response => response.arrayBuffer())
        .then((arrayBuffer) => {
          
        })
      }).catch(err => {
          console.log(err)
      })

**Comment by katspaugh** [comment]
You can call the load() function again like this:

```
wavesurfer.load(url, peaks, duration)

getPeaks().then(res => {
  wavesurfer.load(url, res.data, duration)
})
```

**Reply by code565** [reply]
Mine is a large file, and I want to render the peak first,

**Reply by code565** [reply]
But I don't know how to load the audio in the future

**Reply by katspaugh** [reply]
You want to skip the audio, just pass an empty string for the url like so:

```
wavesurfer.load('', peaks, duration)
```

**Reply by code565** [reply]
Direct rendering of large files will definitely cause browser lag. I would like to ask how to increase audio data more friendly after rendering the peak value

**Reply by katspaugh** [reply]
You could perhaps do something like this:

```js
wavesurfer.load('', initialPeaks, totalDuration)

const allChunks = []
for (let i = 0; i < chunks.length; i++) {
  fetchChunk(i).then(chunk => {
    allChunks.push(chunk)
    const url = URL.createObjectURL(new Blob([allChunks], { type: 'audio/wav' }))
    wavesurfer.load(url, allChunks + emptyPad, totalDuration)
  })
```

I'd like to point out again that wavesufer doesn't support audio concatenation out of the box (and likely will not do so in future), so I cannot offer further support and guidance on this. Please try Stackoverflow.
